best-practices/client-side/framework/react/components/media-cropper/src/components/media-cropper/index.js

```diff
@@ -0,0 +1,204 @@
+            document.removeEventListener('mousemove', this.mouseMove);
+            document.removeEventListener('mouseup', this.mouseUp);
+            document.removeEventListener('mouseleave', this.mouseUp);
+        }
+    }
+
+    click(e) {
+        if (!this.state.moved) {
+            this.refs.fileUploader.click();
+        }
+        this.setState({
+            ...this.state,
+            moved: false
+        })
+    }
+
+    mouseDown(e) {
+        this.setState({
+            ...this.state,
+            dragging: true,
+            mouseStart: {
+                x: e.pageX,
+                y: e.pageY
+            }
+        });
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    mouseMove(e) {
+        this.setState({
+            ...this.state,
+            transX: this.state.transX - (this.state.mouseStart.x - e.pageX),
+            transY: this.state.transY - (this.state.mouseStart.y - e.pageY),
+            mouseStart: {
+                x: e.pageX,
+                y: e.pageY
+            },
+            moved: true
+        });
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    mouseUp(e) {
+        this.setState({
+            ...this.state,
+            dragging: false
+        });
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    mouseWheel(e) {
+        const scale = Math.pow(1.0015, -e.deltaY);
+
+        const mX = e.pageX - e.target.offsetLeft;
+        const mY = e.pageY - e.target.offsetTop;
+
+        let transX = (this.state.transX - mX) * scale + mX;
+        let transY = (this.state.transY - mY) * scale + mY;
+
+        const zoom = this.state.zoom * scale;
+        this.setState({
+            ...this.state,
+            zoom,
+            transX,
+            transY
+        });
+    }
+
+    loadFile(e) {
+        e.preventDefault();
+        e.stopPropagation();
+        const file = e.target.files[0];
+        if (!file) return;
+        if (file.type === 'image/gif') {
+            // TODO: Handle gif
+        } else if (file.type.startsWith('image/')) {
+            const reader = new FileReader();
+
+            reader.onload = () => {
+                const media = new Image();
+                media.src = reader.result;
+                media.setAttribute('crossorigin', 'anonymous');
+
+                media.onload = () => {
+                    const zoom = Math.max(this.props.width / media.width, this.props.height / media.height);
+                    this.setState({
+                        ...this.state,
+                        isVideo: false,
+                        media,
+                        transX: 0,
+                        transY: 0,
+                        zoom,
+                        renderer: ImageRenderer
+                    });
+                    this.forceUpdate();
+                };
+            };
+
+            reader.readAsDataURL(file);
+        } else if (file.type.startsWith('video/')) {
```

Comment:
Are we sure that `file.type` would always start with `video/` or was this specific to this project? If we aren't enforcing that structure, we should be somehow.

Reply:
All images should be type "image/*" and videos should be "video/*" I assumed. do you know of some video files that would have a different type?